DCF_Demo.md

```diff
@@ -0,0 +1,513 @@
+
+// expect outcome: Azure CLI
+{
+  "payload": {},
+  "status": 200
+}
+
+// expected outcome: serial terminal
+
+Receive direct method: dm.1.install
+	Payload: {"package_name":"cipher_v2i1"}
+Create producer for cipher v2i1...
+Interface cipher 1 published with success
+```
+
+Query IPC again to see the new version being installed
+```
+az iot hub invoke-device-method -n [name-of-iothub] -d [name-of-device] --mn "ipc_query.1.query" --mp "{}" 
+
+// expected outcome
+{
+  "payload": {
+    "continuation_token": 655615,
+    "result": [
+      "ipc_query.1",
+      "dm.1",
+      "cipher.1"
+    ]
+  },
+  "status": 200
+}
+```
+
+The same functionality from version 1 still exist, and you can see that the outcome of encrypting the same string "Welcome to Azure IoT!" will do the same output of encryption.
+```
+az iot hub invoke-device-method -n [name-of-iothub] -d [name-of-device] --mn "cipher.1.encrypt" --mp "{\"context\":0, \"src\":\"Welcome to Azure IoT\!\"}" 
+
+// expected outcome
+{
+  "payload": {
+    "dest": "0ZldfV1pbUhhNXhJyTkBEUhhwX2VcbRM="
+  },
+  "status": 200
+}
+```
+
+Now you can use the new context which would use a new alogrithm of the same string message. 
+```
+az iot hub invoke-device-method -n [name-of-iothub] -d [name-of-device] --mn "cipher.1.encrypt" --mp "{\"context\":1, \"src\":\"Welcome to Azure IoT\!\"}" 
+
+// expected outcome: Azure CLI
+{
+  "payload": {
+    "dest": "1P1xYWwQLAQ1ZCVMlAR8UDV0lXRBcNBg="
+  },
+  "status": 200
+}
+
+// expected outcome: serial output
+Receive direct method: cipher.1.encrypt
+	Payload: {"context":1,"src":"Welcome to Azure IoT\\!"}
+```
+
+You can now try the decryption method of this new encrypted message. 
+```
+az iot hub invoke-device-method -n [name-of-iothub] -d [name-of-device] --mn "cipher.1.decrypt" --mp "{\"src\":\"1P1xYWwQLAQ1ZCVMlAR8UDV0lXRAh\"}"
+
+// expected outcome: Azure CLI
+{
+  "payload": {
+    "dest": "Welcome to Azure IoT!"
+  },
+  "status": 200
+}
+// expected outcome: serial output
+Receive direct method: cipher.1.decrypt
+	Payload: {"src":"1P1xYWwQLAQ1ZCVMlAR8UDV0lXRAh"}
+```
+
+</details>
+<br>
+
+<details>
+<summary>Click here for Powershell commands</summary>
+<br>
+
+The commands are different for Powershell because you have to use ` to escape any " double quotes
+
+<br>
+
+Query for existing interfaces on the device 
+```
+az iot hub invoke-device-method -n [name-of-iothub] -d [name-of-device] --mn "ipc_query.1.query" --mp "{}"
+
+// expected outcome
+{
+  "payload": {
+    "continuation_token": 655615,
+    "result": [
+      "ipc_query.1",
+      "dm.1"
+    ]
+  },
+  "status": 200
+}
+
+```
+
+
+Install sprinker interface
+```
+az iot hub invoke-device-method -n [name-of-iothub] -d [name-of-device] --mn "dm.1.install" --mp "{\`"package_name\`":\`"sprinkler_v1i1\`"}"  
+
+// expected outcome
+{
+  "payload": {},
+  "status": 200
+}
```

Comment:
Consider to have only an example of the \`" instead of repeat the entire sequence.